### PR TITLE
[Amazon Bedrock] Add support for function calling reasoning and response

### PR DESCRIPTION
**Pull request by agithomas**
- Enhancement

## Proposed commit message

Add support for guardrail response test and function calling reasoning and response

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist
- Add pipeline test. Ensure success

## How to test this PR locally

- `elastic-package build`
- `elastic-package stack up -v -d --services package-registry`

## Screenshot

image

### CODE REVIEW

**Comment by muthu-mps** [comment]
@agithomas - Can you include guardrail response log in the pipeline tests?

**Comment by agithomas** [comment]
> @agithomas - Can you include guardrail response log in the pipeline tests?

Updated the pipeline tests

File: packages/aws_bedrock/data_stream/invocation/_dev/test/pipeline/test-aws-bedrock.log-expected.json [hunk 1]
```diff
@@ -3799,6 +3799,122 @@
...
+                            },
+                            "stop_reason": "end_turn",
+                            "usage": {
+                                "inputTokens": 15,
+                                "outputTokens": 29,
+                                "totalTokens": 44
+                            }
+                        },
+                        "output_content_type": "application/json",
+                        "output_token_count": 29
```
Comment (muthu-mps) [inline packages/aws_bedrock/data_stream/invocation/_dev/test/pipeline/test-aws-bedrock.log-expected.json:3859]: When looking into the output_token_count, this data is already captured in the `usage.outputTokens` and also in the `gen_ai.usage.completion_token` field. Do you think we need to revisit the redundant field information?

Comment (agithomas) [reply packages/aws_bedrock/data_stream/invocation/_dev/test/pipeline/test-aws-bedrock.log-expected.json:3859]: Probably could be a design decision taken when the datastream was developed. This is beyond the scope of this PR

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Review by kcreddy — APPROVED** [review]

**Pull request merged**